### PR TITLE
Use absolute version numbers to prevent unnecessary breakages

### PR DESCRIPTION
**Pull request by bennlich**
No more carets! No more tildes! No point in being bleeding edge if it means fixing bugs that don't need fixing and reaping nothing in return. Manual upgrade ftw.

### CODE REVIEW

**Comment by bennlich** [comment]
Thoughts on this, Owen?

**Comment by backspaces** [comment]
On Mon, Apr 6, 2015 at 6:34 PM, Benny Lichtner <notifications@github.com>
 wrote:

> Thoughts on this, Owen?
>
You bet!

The TL;DR is: I'd like to stick to Patch updates of our dev-dependencies
(via npm install), within a constant Minor version number. Basically update
to non-breaking versions of our dev-dependencies. I'll modify this with:
except for 0-Major version packages which we'd keep static due to their not
having to follow semver breakage rules.

Details: I've gotten in a lot of discussions recently w/ devs about semver.
Basically I think its impossible even with the best of intensions. Its in
the NP computational class if I'm not mistaken, i.e. the complexity grows
exponentially with the number of dependencies. Some devs are refusing to
follow it for this and other reasons. Coffeescript for example!

Does this complexity matter? Well I think so.  Even AS w/ just 8
dependencies, has 2097 .js files within node_modules!
    ls -R1 node_modules/ | grep .js | grep -v ':' | wc
Hard to believe. I just don't think there any guarantees with that many
files.

Worse, the "0-Major" problem allows *any* semver change to be breaking.
Look at our dependencies:
  "devDependencies": {
    "gulp": "^3.8.5",
    "gulp-coffee": "^2.0.1",
    "gulp-concat": "^2.3.0",
    "gulp-docco": "0.0.4",
    "gulp-rename": "^1.2.0",
    "gulp-task-listing": "^0.3.0",
    "gulp-uglify": "^0.3.1",
    "lazypipe": "^0.2.1"
  }
Notice anything?  :)   Yup, half of our mighty 8 are 0-Major.

Practically speaking, I do like being able to upgrade to the most recent
non-breaking version easily.  To be honest, however, I don't do a fresh npm
install every commit (guilty!). I doubt if many of us do tho.

But! Here's yet another gotcha: npm and node themselves!

I added a "gulp" run script to our package.json thinking I could just then
do "npm run gulp test" to run a test task. Nope.  But when I looked into
it, if I update to a recent npm, I *can* do it via "npm run gulp -- test".
So I updated npm.

So even node and npm versions matter! Damn!

So here's the question: how do we get our wish for updating our dev
dependencies but not breaking?  I think its Patch updates are fine for all
but 0-Major.  Not sure of the syntax.

​And thanks for pointing out the problem .. it started with your finding
out coffee had a breaking update and the gulp-coffee "followed the rules"
by rolling over to a new Minor IIRC.​

**Comment by bennlich** [comment]
K! I switched to ~'s for all package with > 0.x versions.

**Comment by backspaces** [comment]
Nice, thanks.
